src/main/java/org/folio/consortia/service/impl/SharingSettingServiceImpl.java

```diff
@@ -149,6 +164,47 @@ private UUID publishRequest(UUID consortiumId, PublicationRequest publicationReq
     return null;
   }
 
+  private void getFailedSettingsForFailedTenants(UUID consortiumId, UUID publicationId, SharingSettingRequest sharingSettingRequest) {
+    // NOSONAR
+    boolean desiredResultReceived = false;
+    long startTime = System.currentTimeMillis();
+    long timeout = 3000;
+
+    while (Boolean.FALSE.equals(desiredResultReceived) && System.currentTimeMillis() - startTime < timeout) {
+      var publicationStatusEntity = publicationStatusRepository.findById(publicationId);
+      if (publicationStatusEntity.isPresent()) {
+        PublicationDetailsResponse publicationDetails = publicationService.getPublicationDetails(consortiumId, publicationId);
+        if (publicationDetails.getStatus() == PublicationStatus.ERROR) {
+          desiredResultReceived = true;
```

Comment:
if publicationDetails.getStatus() == COMPLETED we also should set desiredResultReceived to true to not wait 3 sec